tusclient/uploader.py

```diff
@@ -211,7 +223,35 @@ def upload_chunk(self):
         """
         Upload chunk of file.
         """
+        self._retried = 0
         self._do_request()
         self.offset = int(self.request.response_headers.get('upload-offset'))
         msg = '{} bytes uploaded ...'.format(self.offset)
         print(msg)
+
+    def _do_request(self):
+        # TODO: Maybe the request should not be re-created everytime.
+        #      The request handle could be left open until upload is done instead.
+        self.request = TusRequest(self)
+        try:
+            self.request.perform()
+            self.verify_upload()
+        except TusUploadFailed as error:
+            self.request.close()
+            self._retry_or_cry(error)
+        finally:
+            self.request.close()
+
+    def _retry_or_cry(self, error):
+        if self.retries > self._retried:
+            time.sleep(self.retry_delay)
```

Comment:
In tus-js-client it works like following: If you start the upload, the actual sending of the request will be performed by the browser meaning that there is no `while(not_finished_uploading) upload_chunk()`-loop. instead the browser just sends events, e.g. when the request finishes or the response is available. This leaves the program unblock and we can just call an abort() method on the current request which will stop it.

For example, here is an example:
```js
let upload = new tus.Upload(...)
upload.start()
setTimeout(function() {
  upload.abort()
}, 1000) // abort after one second
```

Does this seem understandable for you?